packages/react_rearch/lib/src/widgets/consumer.dart

```diff
@@ -0,0 +1,158 @@
+part of '../widgets.dart';
+
+abstract class RearchComponent extends Component2 {
+  bool _building = false;
+
+  String get debugName;
+
+  bool get debug => false;
+
+  void _debug(String msg) {
+    if (!debug) return;
+    // ignore: avoid_print
+    print('<$debugName> -- $msg');
+  }
+
+  final willUnmountListeners = <SideEffectApiCallback>{};
+
+  final sideEffectData = <Object?>[];
+
+  /// Represents a [Set] of functions that remove a dependency on a [Capsule].
+  final dependencyDisposers = <void Function()>{};
+
+  /// Clears out the [Capsule] dependencies of this [RearchComponent].
+  void clearDependencies() {
+    for (final dispose in dependencyDisposers) {
+      dispose();
+    }
+    dependencyDisposers.clear();
+  }
+
+  @override
+  void componentDidMount() {
+    _debug('componentDidMount()');
+    super.componentDidMount();
+  }
+
+  @override
+  void componentDidUpdate(
+    Map<dynamic, dynamic> prevProps,
+    Map<dynamic, dynamic> prevState, [
+    dynamic snapshot,
+  ]) {
+    _debug('componentDidUpdate()');
+    super.componentDidUpdate(prevProps, prevState, snapshot);
+  }
```

Comment:
Don't override these unless you're required to

Reply:
Done